src/DynamoCore/PackageManager/Package.cs

```diff
@@ -348,6 +344,47 @@ private IEnumerable<PackageAssembly> LoadAssembliesInBinDirectory()
             return assemblies;
         }
 
+        /// <summary>
+        ///     Determine if an assembly is in the "node_libraries" list for the package.
+        /// 
+        ///     This algorithm accepts assemblies that don't have the same version, but the same name.
+        ///     This is important when a package author has updated a dll in their package.  
+        /// 
+        ///     This algorithm assumes all of the entries in nodeLibraryFullNames are properly formatted
+        ///     as returned by the Assembly.FullName property.  If they are not, it ignores the entry.
+        /// </summary>
+        /// <param name="nodeLibraryFullNames"></param>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        internal static bool IsNodeLibrary(IEnumerable<string> nodeLibraryFullNames, AssemblyName name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException("name");
+            }
+
+            if (nodeLibraryFullNames == null)
+            {
+                return true;
+            }
+
+            foreach (var n in nodeLibraryFullNames)
+            {
+                try
+                {
+                    // The AssemblyName constructor throws an exception for an improperly formatted string
+                    if (new AssemblyName(n).Name == name.Name)
+                    {
+                        return true;
+                    }
+                }
+                catch (Exception _)
```

Comment:
We really don't want to do anything here? Does this mean I get no indication why my package doesn't load? Does it make sense to write to the console or the log?